docs/source/install_developers.rst

```diff
@@ -0,0 +1,84 @@
+..  _install_developers:
+
+------------------------------
+Installing HDMF for Developers
+------------------------------
+
+Setup
+-----
+
+For development, we recommend installing HDMF in a virtual environment in editable mode. For beginners and for Windows
+developers, we recommend using the `conda package and environment management system`_ for managing local virtual
+environments. Install [Anaconda](https://www.anaconda.com/distribution) to install the ``conda`` tool. You can also use
+the virtualenv_ tool to create a new virtual environment.
+
+.. _conda package and environment management system: https://conda.io/projects/conda/en/latest/index.html
+.. _virtualenv: https://virtualenv.pypa.io/en/stable/
+
+Using conda
+^^^^^^^^^^^
+
+Create and activate a new virtual environment called "hdmf-dev" with Python 3.8 installed.
+
+.. code::
+
+    $ conda create --name hdmf-dev python=3.8
+    $ conda activate hdmf-dev
+
+Using virtualenv
+^^^^^^^^^^^^^^^^
+
+First, create a new virtual environment located at `~/hdmf`.
+
+.. code::
+
+   pip install -U virtualenv
+   virtualenv ~/hdmf
+   source ~/hdmf/bin/activate
+
+Install from GitHub
+-------------------
+
+After you have created and activated a virtual environment, clone the HDMF git repository from GitHub, install the
+package requirements using the pip_ Python package manager, and install HDMF in editable mode.
+
+.. _pip: https://pip.pypa.io/en/stable/
+
+.. code::
+
+   git clone --recurse-submodules https://github.com/hdmf-dev/hdmf.git
+   cd hdmf
+   pip install -r requirements.txt -r requirements-dev.txt -r requirements-doc.txt
+   pip install -e .
+
+Run tests
+---------
+
+You can run the full test suite with the following command:
+
+.. code::
+
+   python test.py
+
+You could also run the full test suite by installing and running the ``pytest`` tool.
+
```

Comment:
Similar to the tox example below it would be nice to add a short code block to show how to run with pytest.